### PR TITLE
Make iter and asyncIter private, add iterable and asyncIterable

### LINKED ISSUE #23: Consider deprecating iter
I think `iter` should probably not be part of the library's public API surface.

Strings and arrays are already iterables. A deprecation notice could warn users when an object is passed to `iter`, informing them that they should use entries instead.

Internally a function called iter could continue to be used with the same basic responsibility of making an iterator out of an iterable.

### PR DESCRIPTION
**Pull request by conartist6**
Fixes #23

### CODE REVIEW

**Comment by conartist6** [comment]
Actually, I originally said deprecate iter and asyncIter, but this diff removes them. There may be enough people using those functions to warrant keeping them around with a deprecation warning.

**Comment by sithmel** [comment]
Using semver and a changelog looks gentle enough. Do you think a deprecation warning is necessary ?

**Comment by conartist6** [comment]
Probably not necessary, but it seems like the polite thing to do. I guess usually the deprecation would go out in a non-major release. It seems like a convenient way of showing people that the new methods are different from the old ones.

**Comment by conartist6** [comment]
Sorry I keep tacking on commits. I'm done now.

**Comment by sithmel** [comment]
ok, I am going to review and merge ... as soon as I have 10 minutes

File: src/internal/iterable.mjs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const emptyArr = []
+export default function iterable (i, ...args) {
```
Comment (sithmel) [inline src/internal/iterable.mjs]: we can remove args here

File: src/internal/async-iterable.mjs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import iterable from './iterable'
+
+export default async function * asyncIterable (i, ...args) {
+  if (i && i[Symbol.asyncIterator]) {
```
Comment (sithmel) [inline src/internal/async-iterable.mjs]: we can remove args here

**Comment by conartist6** [comment]
Renamed. Removed unnecessary ...args. Fixed a bug in asyncZip.

**Pull request merged**